test/test_helper.rb

```diff
@@ -65,6 +65,7 @@ def headless_chrome_driver
 class ActionDispatch::IntegrationTest
   setup { host! Gemcutter::HOST }
 end
+Capybara.server = :webrick
```

Comment:
```ruby
Capybara.server = :unicorn
```

works as well. That way it is closer to production and webrick is not needed at all if I understand it well. On the other side it is not really related change to Ruby 3. I can submit PR later.

Reply:
Is unicorn a supported capybara server? I am getting following error:
```
Error:                                                                                                                                                
StatsTest#test_downloads_animation_bar:                                                                                                               
NoMethodError: undefined method `call' for nil:NilClass                                                                                               
    /home/aditya/.rvm/gems/ruby-2.7.4/gems/capybara-3.35.3/lib/capybara/server.rb:77:in `block in boot' 
```
check: #2769